src/aleph/http/server.clj

```diff
@@ -378,6 +394,21 @@
       ([_ ctx msg]
         (cond
 
+          ;; Happens when io.netty.handler.codec.http.HttpObjectAggregator is part of the pipeline.
+          (instance? FullHttpRequest msg)
+          (if (invalid-request? msg)
+            (reject-invalid-request ctx msg)
+            (let [^FullHttpRequest req msg
+                  content (.content req)
+                  s (netty/source (netty/channel ctx))]
+              (when-not (zero? (.readableBytes content))
+                ;; Retain the content of FullHttpRequest one extra time to
+                ;; compensate for it being released together with the request.
+                (.retain content)
+                (netty/put! (.channel ctx) s content))
```

Comment:
nit: you took channel twice, using different APIs

Reply:
another nit: there's `netty/acquire` wrapper for `.retain` operations, you can do

```clojure
(netty/put! (.channel ctx) s (netty/acquire content))
```

Reply:
👌